test/e2e/test_job_submissions.py

```diff
@@ -61,6 +61,107 @@ def test_success(
+            sleep 90
+            """,
+        )
+
+        try:
+
+            @backoff.on_predicate(
+                wait_gen=backoff.constant,
+                max_time=120,
+                interval=10,
+            )
+            def is_job_started(current_job: Job) -> bool:
+                current_job.refresh_job_info(client=deadline_client)
+                LOG.info(f"Waiting for job {current_job.id} to be created")
+                return current_job.lifecycle_status != "CREATE_IN_PROGRESS"
+
+            assert is_job_started(job)
+
+            @backoff.on_predicate(backoff.constant, interval=5, max_time=60)
+            def sessions_exist(current_job: Job) -> bool:
+                sessions: list[dict[str, Any]] = deadline_client.list_sessions(
+                    farmId=current_job.farm.id, queueId=current_job.queue.id, jobId=current_job.id
+                ).get("sessions")
+
+                return len(sessions) > 0
+
+            assert sessions_exist(job)
+
+            queue_credentials_directory = f"/var/lib/deadline/queues/{job.queue.id}"
+
+            # Verify that the queue user is able to access the credentials file
+            check_queue_user_can_access_credentials_result = session_worker.send_command(
+                command=f"sudo -u {posix_job_user.user} [ -e '{queue_credentials_directory}/aws_credentials.json' ]"
+            )
+            assert check_queue_user_can_access_credentials_result.exit_code == 0
+
+            # Verify that any other users are not able to access the credential files
+
+            check_other_user_cannot_access_credentials_result = session_worker.send_command(
+                command=f"sudo -u {generic_non_queue_job_user.user} [ -e '{queue_credentials_directory}/aws_credentials.json' ]"
+            )
+
+            assert check_other_user_cannot_access_credentials_result.exit_code != 0
+
+            # Verify that another queue's user cannot access the credentials file through a job
+            second_queue_job = submit_custom_job(
+                "Test Getting Primary Queue Credentials File",
+                deadline_client,
+                deadline_resources.farm,
+                deadline_resources.queue_b,
+                f"""
+                #!/usr/bin/env bash
+                cat {queue_credentials_directory}/aws_credentials.json
+                """,
+                max_retries_per_task=0,
+            )
+            try:
+                second_queue_job.wait_until_complete(client=deadline_client)
+                assert second_queue_job.task_run_status == TaskStatus.FAILED
+
+            finally:
+                deadline_client.update_job(
+                    farmId=second_queue_job.farm.id,
+                    queueId=second_queue_job.queue.id,
+                    jobId=second_queue_job.id,
+                    targetTaskRunStatus="CANCELED",
+                )
+                second_queue_job.wait_until_complete(client=deadline_client)
```

Comment:
Can this be made into a separate test case? These tests take a long time to run so it would be great if our assertions/test cases are more fine grained, especially if we need to submit another job to run this test case anyway

Reply:
Makes sense, okay, I'll add a separate test case.